# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/DOMSecurityManager.h
# Commit: 8705284b50d4
# Full Hash: 8705284b50d4905110df9b1fab7f00d9d77d06e6
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-10-22 15:53:00
# Regressor Bug: 1584993
# File Overlap Count: 1
# Description:
#   Bug 1584993: Make CSP frame-ancestors work with fission enabled. r=jkt,farre,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49147
# ==============================================================================

diff -r 141755ff04e1 -r 8705284b50d4 dom/security/DOMSecurityManager.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/security/DOMSecurityManager.h	Tue Oct 22 08:53:47 2019 +0000
@@ -0,0 +1,31 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_DOMSecurityManager_h
+#define mozilla_dom_DOMSecurityManager_h
+
+#include "nsIObserver.h"
+
+class DOMSecurityManager final : public nsIObserver {
+ public:
+  NS_DECL_ISUPPORTS
+  NS_DECL_NSIOBSERVER
+
+  static void Initialize();
+
+ private:
+  DOMSecurityManager() = default;
+  ~DOMSecurityManager() = default;
+
+  // Only enforces the frame-anecstor check which needs to happen in
+  // the parent because we can only access the window global in the
+  // parent. The actual CSP gets parsed and applied in content.
+  nsresult ParseCSPAndEnforceFrameAncestorCheck(nsIChannel* aChannel);
+
+  static void Shutdown();
+};
+
+#endif /* mozilla_dom_DOMSecurityManager_h */